stl/inc/span

```diff
@@ -0,0 +1,679 @@
+    constexpr span(array<value_type, _Size>& _Arr) noexcept : _Mybase(_Size), _Mydata(_Arr.data()) {}
+
+    template <size_t _Size, size_t _Ext = _Extent,
+        enable_if_t<conjunction_v<bool_constant<_Ext == dynamic_extent || _Ext == _Size>,
+                        is_convertible<const value_type*, element_type*>>,
+            int> = 0>
+    constexpr span(const array<value_type, _Size>& _Arr) noexcept : _Mybase(_Size), _Mydata(_Arr.data()) {}
+
+    template <class _Cont, enable_if_t<conjunction_v<bool_constant<_Extent == dynamic_extent>,
+                                           _Is_span_compatible_container<_Cont, element_type>>,
+                               int> = 0>
+    constexpr span(_Cont& _Container)
+        : _Mybase(static_cast<index_type>(_STD size(_Container))), _Mydata(_STD data(_Container)) {}
+
+    template <class _Cont, enable_if_t<conjunction_v<bool_constant<_Extent == dynamic_extent>,
+                                           _Is_span_compatible_container<const _Cont, element_type>>,
+                               int> = 0>
+    constexpr span(const _Cont& _Container)
+        : _Mybase(static_cast<index_type>(_STD size(_Container))), _Mydata(_STD data(_Container)) {}
+
+    template <class _OtherTy, size_t _OtherExtent,
+        enable_if_t<conjunction_v<bool_constant<_Extent == dynamic_extent || _Extent == _OtherExtent>,
+                        is_convertible<_OtherTy (*)[], element_type (*)[]>>,
+            int> = 0>
+    constexpr span(const span<_OtherTy, _OtherExtent>& _Other) noexcept
+        : _Mybase(_Other.size()), _Mydata(_Other.data()) {}
+
+    // [span.sub] Subviews
+    template <size_t _Count>
+    _NODISCARD constexpr span<element_type, _Count> first() const noexcept /* strengthened */ {
+        if constexpr (_Extent != dynamic_extent) {
+            static_assert(_Count <= _Extent, "Count out of range in span::first()");
+        }
+#if _CONTAINER_DEBUG_LEVEL > 0
+        else {
+            _STL_VERIFY(_Count <= size(), "Count out of range in span::first()");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {_Mydata, _Count};
+    }
+
+    _NODISCARD constexpr span<element_type, dynamic_extent> first(const index_type _Count) const noexcept
+    /* strengthened */ {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Count <= size(), "Count out of range in span::first(count)");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {_Mydata, _Count};
+    }
+
+    template <size_t _Count>
+    _NODISCARD constexpr span<element_type, _Count> last() const noexcept /* strengthened */ {
+        if constexpr (_Extent != dynamic_extent) {
+            static_assert(_Count <= _Extent, "Count out of range in span::last()");
+        }
+#if _CONTAINER_DEBUG_LEVEL > 0
+        else {
+            _STL_VERIFY(_Count <= size(), "Count out of range in span::last()");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {_Mydata + size() - _Count, _Count};

```

Comment:
The standard depicts parens around `size() - _Count` I think you should keep to avoid 2 pointer adjustments instead of 1.

Reply:
Done, 

Evaluation ordering is hard :(